apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.jsx

```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import ActivityCard from '@cdo/apps/lib/levelbuilder/lesson-editor/ActivityCard';
+import Activity from '@cdo/apps/templates/lessonOverview/activities/Activity';
+import {connect} from 'react-redux';
+import {addActivity} from '@cdo/apps/lib/levelbuilder/lesson-editor/activitiesEditorRedux';
+
+const styles = {
+  activityEditAndPreview: {
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  editor: {
+    width: '50%'
+  },
+  preview: {
+    width: '45%',
+    marginLeft: 20
+  },
+  previewBox: {
+    border: '1px solid black',
+    padding: 10
+  },
+  addActivity: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none'
+  }
+};
+
+class ActivitiesEditor extends Component {
+  static propTypes = {
+    //redux
+    activities: PropTypes.array,
+    addActivity: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      targetActivitySectionPos: null
+    };
+  }
+
+  handleAddActivity = () => {
+    this.props.addActivity(
+      this.props.activities.length,
+      `activity-${this.generateActivityKey()}`
```

Comment:
on second read, please move the prefix to be part of what gets returned by the generate method. otherwise there is a fragile dependency that you're adding the right prefix to whatever this method returns. (here and for other generate methods)